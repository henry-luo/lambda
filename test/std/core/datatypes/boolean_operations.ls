// Test: Boolean Operations
// Category: core/datatypes
// Type: positive
// Expected: 1

// Boolean literals
let t = true
let f = false

// Logical operations
let and1 = t && t    // true
let and2 = t && f     // false
let or1 = t || f      // true
let or2 = f || f      // false
let not_t = !t        // false
let not_f = !f        // true

// Short-circuit evaluation
let short_circuit = f && (1/0 == 0)  // false (doesn't evaluate division)

// Comparison operations
let eq = t == true    // true
let neq = t != false  // true

// Type conversion in comparisons
let truthy = t == 1    // true (if 1 is truthy)
let falsy = f == 0     // true (if 0 is falsy)

// Final check
1
