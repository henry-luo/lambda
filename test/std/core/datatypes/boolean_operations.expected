true
false
true
false
true
false
false
true
true
true
true
false
